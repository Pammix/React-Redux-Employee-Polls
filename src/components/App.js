import { useEffect, Fragment } from 'react';
import Login from './Login';
import { connect } from 'react-redux';
import { handleInitialData } from '../actions/shared';
import Dashboard from './Dashboard';
import Leaderboard from './Leaderboard';
import NewPoll from './NewPoll';
import NotFoundPage from './NotFoundPage';
import Nav from './Nav';
import { Routes, Route } from 'react-router-dom';
import Poll from './Poll';

const App = (props) => {
  useEffect(() => {
    if (props.loggedIn) {
      return;
    }
    props.dispatch(handleInitialData());
  });

  if (!props.loggedIn) {
    return <Login/>;
  }
  return (
    <Fragment>
      <div className='container'>
        {props.loggedIn && <Nav />}
        <Routes>
          <Route path='/login' exact element={<Login />} />
          <Route path='/' element={<Dashboard />} />
          <Route path='/questions/:id' element={<Poll />} />
          <Route path='/add' exact element={<NewPoll />} />
          <Route path='/board' exact element={<Leaderboard />} />
          <Route path='/404' exact element={<NotFoundPage />} />
        </Routes>
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authUser }) => {
  return { loggedIn: authUser != null };
};

export default connect(mapStateToProps)(App);
