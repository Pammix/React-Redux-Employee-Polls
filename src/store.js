import { configureStore } from '@reduxjs/toolkit';
import authUser from './reducers/authUser';
import users from './reducers/users';
import questions from './reducers/questions';
import logger from './middleware/logger';
import thunk from 'redux-thunk';

export const store = configureStore({
  reducer: {
    authUser,
    users,
    questions
  },
  middleware: [thunk, logger]
});
