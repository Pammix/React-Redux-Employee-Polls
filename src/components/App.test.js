import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import React from 'react';
import { store } from '../store';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import App from './App';

describe('App', () => {
  it('should render the component', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });
  it('should render the Login component', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(view.queryByTestId('login-head')).toBeInTheDocument();
  });
});
