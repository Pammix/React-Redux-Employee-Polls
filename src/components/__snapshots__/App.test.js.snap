// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-wrapper"
        data-testid="login-head"
      >
        <h1
          class="gradient-text"
        >
          Employee Polls
        </h1>
        <figure>
          <img
            alt="Poll"
            src="poll-logo.png"
          />
        </figure>
        <h3>
           Log In 
        </h3>
        <form
          id="loginForm"
        >
          <label>
            <p>
              User
            </p>
            <input
              data-testid="username"
              type="text"
              value="sarahedo"
            />
          </label>
          <label>
            <p>
              Password
            </p>
            <input
              data-testid="password"
              id="password"
              type="password"
            />
          </label>
          <div
            class="button-container"
          >
            <input
              data-testid="submit-btn"
              id="btnlogin"
              type="submit"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-wrapper"
      data-testid="login-head"
    >
      <h1
        class="gradient-text"
      >
        Employee Polls
      </h1>
      <figure>
        <img
          alt="Poll"
          src="poll-logo.png"
        />
      </figure>
      <h3>
         Log In 
      </h3>
      <form
        id="loginForm"
      >
        <label>
          <p>
            User
          </p>
          <input
            data-testid="username"
            type="text"
            value="sarahedo"
          />
        </label>
        <label>
          <p>
            Password
          </p>
          <input
            data-testid="password"
            id="password"
            type="password"
          />
        </label>
        <div
          class="button-container"
        >
          <input
            data-testid="submit-btn"
            id="btnlogin"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
